lib/mongo/protocol/message.rb

```diff
@@ -190,9 +191,7 @@ def hash
       #   server. The server will put this id in the response_to field of
       #   a reply.
       def set_request_id
-        @@id_lock.synchronize do
-          @request_id = @@request_id += 1
-        end
+        @request_id = Message.next_id
```

Comment:
`Message.next_id` should be spelled `self.class.next_id` to use the correct class.